.github/workflows/update-public-docs.yml

```
@@ -59,31 +62,90 @@ jobs:
           prerelease: false
 
       - name: Upload Release Asset
-        if: ${{ steps.check_docs.outputs.DOCS_CHANGED }}
-        id: upload-release-asset
+        if: ${{ steps.check_release_changed.outputs.DOCS_CHANGED }}
+        id: upload_release_asset
         uses: actions/upload-release-asset@v1
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          upload_url: ${{ steps.create-release.outputs.upload_url }}
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
           asset_path: ./new_public_docs.json
           asset_name: public_docs.json
           asset_content_type: application/json
 
-      - name: Trigger update in docs repo
-        if: ${{ steps.check_docs.outputs.DOCS_CHANGED }}
+      - name: Clone docs repo
+        if: ${{ steps.check_release_changed.outputs.DOCS_CHANGED }}
+        uses: actions/checkout@v3
+        with:
+          repository: opensafely/documentation
+          path: docs
+          ref: 'main'
+          token: ${{ secrets.DOCS_WRITE_TOKEN }}
+
+      - name: Check if update branch exists
+        if: ${{ steps.check_release_changed.outputs.DOCS_CHANGED }}
+        id: check-branch
+        run: |
+          cd docs
+          echo ::set-output name=branch-exists::`git ls-remote --heads origin ${BRANCH}`
+
+      - name: Create or checkout branch
+        if: ${{ steps.check_release_changed.outputs.DOCS_CHANGED }}
+        run: |
+          cd docs
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          if [ -n "${{ steps.check-branch.outputs.branch-exists }}" ]; then
```

I think you could get rid of the previous step by embedding the check from it here
```suggestion
          if git ls-remote --heads origin ${BRANCH} >/dev/null 2>&1; then
```